src/components/AppBody/index.tsx

```diff
@@ -41,31 +44,49 @@ const AppBody = () => {
     }
   };
 
+  const constructUrlWithQueryParams = () => {
+    if (queryParams.length == 0) {
+      return url;
+    }
+    const queryParamStrings = queryParams.map(
+      (queryParam: QueryParam) => `${queryParam.key}=${queryParam.value}`
+    );
+    const combinedQuery = queryParamStrings.join('&');
+    return `${url}?${combinedQuery}`;
+  };
+
   const onSend = () => {
```

Comment:
Can we refactor this onSend() method? I feel the combinedUrl, setIsLoading, setIsNoRequestTriggered is the same here for all HTTP method and can be defined at the start of the method.

Reply:
Yes Nithin that would be better and we can avoid the duplication here. 
I will make the changes.

Reply:
This looks good!